solver/src/main/java/org/chocosolver/util/tools/RealUtils.java

```diff
@@ -729,4 +731,24 @@ public static RealInterval acos_wrt(RealInterval i, RealInterval res) {
         return new RealIntervalConstant(retInf, retSup);
     }
 
+    public static RealInterval updateBounds(RealInterval var, RealInterval newInterval) {
+        return updateBounds(var, newInterval.getLB(), newInterval.getUB());
+    }
+
+    public static RealInterval updateBounds(RealInterval var, double lb, double ub) {
+        return new RealIntervalConstant(
+                Math.max(var.getLB(), lb),
+                Math.min(var.getUB(), ub));
+    }
+
+    public static double roundValue(double value, double precision) {
```

Comment:
That seems strange to me to pass by a `String` to get a double.
I think calling `prevFloat` and `nextFloat` are enough in `isEntailed`